Add new arguments to by_id()

KuhnMn opened the pull request:
Made all arguments from the materials project side available in the by_id() method. I was unsure of how to add the new optional Union return type to the docstring, so a quick check here would be appreciated! This closes pyiron/pyiron_atomistics#950.

KuhnMn committed:
Add new arguments to by_id()

Made all arguments from the materials project side available in the by_id() method. This was proposed in #950.

pmrv commented on pyiron_atomistics/atomistics/structure/factories/materialsproject.py:
```diff
@@ -139,9 +144,14 @@ def by_id(material_id: Union[str, int], api_key=None) -> Atoms:
         Returns:
-            :class:`~.Atoms`: requested structure
+            :class:`~.Atoms`: requested final structure if final is True or
+            a list of initial structures if final is False
```
```suggestion
            :class:`~.Atoms`: requested final structure if final is True
            list of :class:~.Atoms`:  a list of initial (pre-relaxation) structures if final is False
```

is probably how I would do it,  but no hard feelings. Thanks for this timely PR!

pmrv reviewed (approved):
lgtm!

KuhnMn committed:
:memo: Update docstring of return type

Update the docstring of the return type as suggested on the PR.

Co-authored-by: Marvin Poul <ponder@creshal.de>

KuhnMn commented:
Thanks for the suggestion, I thought it looked cleaner than mine so I applied it! I'm not sure why the workflows are failing since when looking at the logs, the file I changed is not mentioned at all :/

samwaseda commented on pyiron_atomistics/atomistics/structure/factories/materialsproject.py:
```diff
@@ -152,4 +162,18 @@ def by_id(material_id: Union[str, int], api_key=None) -> Atoms:
+                    pymatgen_to_pyiron(mpr_structure)
+                    for mpr_structure in (
+                        mpr.get_structure_by_material_id(
+                            material_id, final, conventional_unit_cell
```
```suggestion
                            material_id=material_id, final=final, conventional_unit_cell=conventional_unit_cell
```

Whenever we use an external tool, it's always good to set the arguments explicitly, because they might change the order or add/remove certain arguments and we might not notice it.

KuhnMn committed:
Explicitly set arguments in external method call

Follow samwaseda's advice to explicity set the arguments when calling the materials project  method.

Co-authored-by: Sam Dareska <37879103+samwaseda@users.noreply.github.com>

KuhnMn commented:
Thanks for the advice, I applied your suggestion!

samwaseda commented on pyiron_atomistics/atomistics/structure/factories/materialsproject.py:
```diff
@@ -152,4 +162,18 @@ def by_id(material_id: Union[str, int], api_key=None) -> Atoms:
+            if final:
+                return pymatgen_to_pyiron(
+                    mpr.get_structure_by_material_id(
+                        material_id, final, conventional_unit_cell
```
```suggestion
                        material_id=material_id, final=final, conventional_unit_cell=conventional_unit_cell
```

Here as well

KuhnMn committed:
Explicity set arguments in other method call

Co-authored-by: Sam Dareska <37879103+samwaseda@users.noreply.github.com>

KuhnMn commented:
My bad, I was looking at github on my mobile device and didn't notice that the other call was not changed! Thanks for doing it for me, I applied the suggestion!

samwaseda reviewed (approved):
ok looks good to me

pmrv commented:
It seems the error in the unit test is unrelated to the changes here.  Might be the `sqlalchemy` problems we had a while ago.  @KuhnMn mind to rebase your changes onto the current main branch?

KuhnMn committed:
:art: Reformat using Black

pmrv commented:
Thanks for contributing @KuhnMn !
